src/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTraceSymbols.CoreCLR.cs

```diff
@@ -123,20 +126,25 @@ private unsafe MetadataReader TryGetReader(string assemblyPath, IntPtr loadedPeA
             MetadataReaderProvider provider;
             if (_metadataCache.TryGetValue(cacheKey, out provider))
             {
+                if (provider == null)
+                {
+                    return null;
+                }
                 return provider.GetMetadataReader();
             }
 
             provider = (inMemoryPdbAddress != IntPtr.Zero) ?
                 TryOpenReaderForInMemoryPdb(inMemoryPdbAddress, inMemoryPdbSize) :
                 TryOpenReaderFromAssemblyFile(assemblyPath, loadedPeAddress, loadedPeSize);
 
+            // This may fail as another thread might have beaten us to it, but it doesn't matter
+            _metadataCache.TryAdd(cacheKey, provider);
```

Comment:
We want to add it even if provider is null?

Reply:
Yep : )
Null means no symbols were found for a module so its a common case in this code path and very valuable performance-wise to cache.